include/openssl/core_dispatch.h

```diff
@@ -951,6 +952,10 @@ OSSL_CORE_MAKE_FUNC(int, store_close, (void *loaderctx))
 OSSL_CORE_MAKE_FUNC(int, store_export_object,
                     (void *loaderctx, const void *objref, size_t objref_sz,
                      OSSL_CALLBACK *export_cb, void *export_cbarg))
+OSSL_CORE_MAKE_FUNC(void *, store_open_ex,
+                    (void *provctx, const char *uri,
+                     const char *propq, const OSSL_PARAM params[],
+                     OSSL_PASSPHRASE_CALLBACK *pw_cb, void *pw_cbarg))
```

Comment:
As an asnwer to https://github.com/openssl/openssl/pull/20131/files#r1231808173:

The `propq` argument here should be removed.
The added `pw_cb` and `pw_cbarg` should stay, as a password mechanism was there in the corresponding legacy function signature (for engine implementations):

https://github.com/openssl/openssl/blob/2525109f90cf3a91a909621266ec6854a83805e2/include/openssl/store.h#L287-L289

The added `params`...  I view as an option.  Not really necessary in my mind, but we have added that as a convenience with other operations' constructors, so...

So this signature should change to this:

``` C
OSSL_CORE_MAKE_FUNC(void *, store_open_ex,
                    (void *provctx, const char *uri,
                     OSSL_PASSPHRASE_CALLBACK *pw_cb, void *pw_cbarg))
```

or this:

``` C
OSSL_CORE_MAKE_FUNC(void *, store_open_ex,
                    (void *provctx, const char *uri, const OSSL_PARAM params[],
                     OSSL_PASSPHRASE_CALLBACK *pw_cb, void *pw_cbarg))
```

@simo5, I leave that final choice to you, and will accept either.

Reply:
Thanks I will make the change likely next week.